### PR TITLE
Also run sanitize on the content when converting from xml to json.

### PR DESCRIPTION
**Pull request by ehoogerbeets**
Previously, the content was unsanitized when the data was converted from xml to json, but not sanitized when converted from json to xml. This is corrected now so that the conversion is reversible.

### CODE REVIEW

**Comment by ehoogerbeets** [comment]
@c4milo Please have a look at this new PR for you in xml2json. I did this change because I am representing HTML strings in an XLIFF file (XML Localization Interchange File Format) and the embedded HTML tags were screwing up the XML syntax without this sanitizing.

**Comment by c4milo** [comment]
Nice, thanks for this, I'll review tomorrow.

**Comment by ehoogerbeets** [comment]
@c4milo review?

File: lib/xml2json.js [hunk 1]
```diff
@@ -60,9 +60,10 @@ function endElement(name) {
             currentObject['$t'] = currentObject['$t'].trim()
         }
 
-        if (options.sanitize) {
-            currentObject['$t'] = sanitizer.sanitize(currentObject['$t'], true);
-        }
+        // node-expat already reverse sanitizes it whether we like it or not
```
Comment (c4milo) [inline lib/xml2json.js]: I would just remove these comments, leaving them in the commit message instead.

**Comment by c4milo** [comment]
Apologies it took me so long, it looks good to me in general. I Only commented on a minor thing.

File: lib/json2xml.js [hunk 2]
```diff
@@ -76,13 +76,13 @@ ToXml.prototype.openTag = function(key) {
...
     if (this.options.sanitize) {
-        val = sanitizer.sanitize(val)
+        val = sanitizer.sanitize(val);
     }
     this.xml += ' ' + key + '="' + val + '"';
 }
 ToXml.prototype.addTextContent = function(text) {
     this.completeTag();
-    this.xml += text;
+    this.xml += (this.options.sanitize ? sanitizer.sanitize(text) : text);
```
Comment (c4milo) [inline lib/json2xml.js]: Please use multiple lines so a step by step debugging can be possible.

**Comment by antti-manninen** [comment]
Is this pull request still alive? This would be really useful fix.

**Comment by ehoogerbeets** [comment]
Hi Anti, I didn't see c4milo's response to this. I'll work on it now and resolve the merge conflict.

File: test/test.js [hunk 3]
```diff
@@ -99,14 +99,56 @@ describe('xml2json', function () {
...
+        done();
+    });
+
+    it('does doesnt double unsanitize', function (done) {
+
+        var xml = internals.readFixture('xmlsanitize3.xml');
+        var result = parser.toJson(xml, {sanitize: true, reversible: true});
+        var json = internals.readFixture('xmlsanitize3.json');
+
+        expect(result).to.equal(json);
```
Comment (antti-manninen) [inline test/test.js]: Travis is failing because done() is missing from this test case.

**Comment by ehoogerbeets** [comment]
Okay, ready to merge.

**Comment by antti-manninen** [comment]
c4milo, can you publish these changes to npm also?

**Pull request merged**